Enhancement: Consul Compatibility Checking (#15818)
hashicorp/consul: website/content/commands/connect/envoy.mdx

  TLS on the Prometheus metrics endpoint. Only applicable when `envoy_prometheus_bind_addr`
  is set in proxy config.

> - `-ignore-envoy-compatibility` - Specifies whether or not the program should check the envoy version for compatibility.
> If this flag is set to true, the program will not check the envoy version for compatibility.
> It is recommended in most situations to leave this flag set to false, as ensuring compatibility with the
> Envoy version can prevent potential issues. By default this is set to `false`.

Reviewer: ```suggestion
- `-ignore-envoy-compatibility` - If set to `true`, this flag ignores the Envoy version compatibility check. We recommend setting this flag set to `false` to ensure compatibility with Envoy and prevent potential issues. Default is `false`.
```
Just a few tweaks to simplify the description and consistently use monospace. Did we update the documentation under /connect/proxies/envoy or wherever we describe the Envoy startup process? I was going to add a link, but I'm not sure where it would go. Maybe this would be a better place to document this behavior? /connect/registration/service-registration

Author: Sorry adam, I'm not sure which paths you're referring to? Where exactly is /connect/registration/service-registration or /connect/proxies/envoy?

This is described here because this is a flag for and specific to the `consul connect envoy` command

Reviewer: I was thinking one of these pages would be the place to describe the default bootstrap process: 
https://developer.hashicorp.com/consul/docs/connect/proxies/envoy
https://developer.hashicorp.com/consul/docs/connect/registration/service-registration

Maybe we don't need it, though. The beginning of this page says "The default behavior is to generate the necessary bootstrap configuration for Envoy based on the environment variables and options provided and by talking to the local Consul agent." Maybe we should add a sentence there letting people know about the version check.

Reviewer 2: Do we need to add a sentence about the version check at an overview level?

It seems like a thing that you don't need to be told about. If you run into it (because you're using an incompatible version), the error message you get will tell you what went wrong and why. It seems like something that's already very discoverable in-context in Consul itself, so doesn't need to be present at an overview level in the docs... Freeing us up to focus on other important, high-level info in the docs.

Author: I think that makes sense to me

Author: @jkirschner-hashicorp and @trujillo-adam are you okay with me resolving this discussing and merging this PR?

Author: I'm going to close this PR for now, we can always create another one if we want to address the documentation updates
